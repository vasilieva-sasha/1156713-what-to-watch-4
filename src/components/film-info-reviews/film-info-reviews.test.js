import React from "react";
import renderer from "react-test-renderer";
import FilmInfoReviews from "./film-info-reviews";
import configureStore from 'redux-mock-store';
import NameSpace from './../../reducer/name-space';
import {Provider} from "react-redux";

const mock = [{
  id: 0,
  user: {
    id: 0,
    name: ``,
  },
  comment: ``,
  date: `PropTypes.string.isRequired`,
  rating: 0,
}];

const mockStore = configureStore([]);

describe(`ReviewsComponent`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      currentPage: `INFO`,
    },
    [NameSpace.DATA]: {
      reviews: mock,
      serverError: false
    },
  });
  it(`film reviews component renders correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <FilmInfoReviews film={mock} currentPage={store.currentPage} serverError={store.serverError}/>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`film reviews component renders correctly with server error`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <FilmInfoReviews film={mock} currentPage={store.currentPage} serverError={true}/>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
