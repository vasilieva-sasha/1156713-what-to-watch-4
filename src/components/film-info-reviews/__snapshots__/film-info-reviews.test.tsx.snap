// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsComponent film reviews component renders correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            
          </cite>
          <time
            className="review__date"
            dateTime="2020-07-09T16:06:01.831Z"
          >
            July 09, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        0
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  />
</div>
`;

exports[`ReviewsComponent film reviews component renders correctly with server error 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            
          </cite>
          <time
            className="review__date"
            dateTime="2020-07-09T16:06:01.831Z"
          >
            July 09, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        0
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  />
</div>
`;
